src/creature.cpp

```diff
@@ -762,6 +762,10 @@ void Creature::changeHealth(int32_t healthChange, bool sendHealthChange/* = true
 	if (sendHealthChange && oldHealth != health) {
 		g_game.addCreatureHealth(this);
 	}
+
+	if (health <= 0) {
+		g_dispatcher.addTask(createTask(std::bind(&Game::executeDeath, &g_game, getID())));
```

Comment:
This method (changeHealth) is always executed in dispatchee thread anyway, so I guess you can just call that method directly?